clinica/pydra/statistics_volume/task.py

```diff
@@ -0,0 +1,633 @@
+                "[Error] output folder " + dirname(spm_mat) + " does not exist."
+            )
+        else:
+            raise RuntimeError("[Error] SPM matrix " + spm_mat + " does not exist.")
+
+    list_files = [f for f in listdir(dirname(spm_mat)) if not f.startswith(".")]
+
+    # Handle figure files
+    figures = [
+        abspath(join(dirname(spm_mat), f)) for f in list_files if f.endswith("png")
+    ]
+    if len(figures) < 2:
+        raise RuntimeError("[Error] Figures were not generated")
+    fig_number = [int(f[-7:-4]) for f in figures]
+    spm_figures = [
+        abspath("./" + "group-" + group_label + "_report-" + str(i) + ".png")
+        for i in fig_number
+    ]
+    for old_name, new_name in zip(figures, spm_figures):
+        copyfile(old_name, new_name)
+
+    # Handle spm t maps
+    spm_T = [
+        abspath(join(dirname(spm_mat), f)) for f in list_files if f.startswith("spmT")
+    ]
+    spm_T = sorted(spm_T)
+    if len(spm_T) != 2:
+        raise RuntimeError("[Error] " + str(len(spm_T)) + " SPM t-map(s) were found")
+    if fwhm:
+        spmT_0001 = abspath(
+            f"group-{group_label}_{class_names[0]}-lt-{class_names[1]}_measure-{measure}_fwhm-{int(fwhm)}_TStatistics.nii"
+        )
+        spmT_0002 = abspath(
+            f"group-{group_label}_{class_names[1]}-lt-{class_names[0]}_measure-{measure}_fwhm-{int(fwhm)}_TStatistics.nii"
+        )
+    else:
+        spmT_0001 = abspath(
+            f"group-{group_label}_{class_names[0]}-lt-{class_names[1]}_measure-{measure}_TStatistics.nii"
+        )
+        spmT_0002 = abspath(
+            f"group-{group_label}_{class_names[1]}-lt-{class_names[0]}_measure-{measure}_TStatistics.nii"
+        )
+    copyfile(join(dirname(spm_mat), "spmT_0001.nii"), spmT_0001)
+    copyfile(join(dirname(spm_mat), "spmT_0002.nii"), spmT_0002)
+
+    variance_of_error = abspath("./group-" + group_label + "_VarianceError.nii")
+    copyfile(abspath(join(dirname(spm_mat), "ResMS.nii")), variance_of_error)
+
+    resels_per_voxels = abspath("./resels_per_voxel.nii")
+    copyfile(abspath(join(dirname(spm_mat), "RPV.nii")), resels_per_voxels)
+
+    mask = abspath("./included_voxel_mask.nii")
+    copyfile(abspath(join(dirname(spm_mat), "mask.nii")), mask)
+
+    # Handle beta files
+    betas = [
+        abspath(join(dirname(spm_mat), f)) for f in list_files if f.startswith("beta_")
+    ]
+    betas = sorted(betas)
+    if len(betas) != 2 + len(covariates):
+        raise RuntimeError("[Error] Not enough betas files found in output directory")
+    regression_coeff_covar = [abspath("./" + covar + ".nii") for covar in covariates]
+    regression_coeff = [
+        abspath("./" + class_names[0] + ".nii"),
+        abspath("./" + class_names[1] + ".nii"),
+    ]
+    regression_coeff.extend(regression_coeff_covar)
+    # Order is respected:
+    for beta, reg_coeff in zip(betas, regression_coeff):
+        copyfile(beta, reg_coeff)
+
+    # Handle contrast files
```

Comment:
All these comments breaking this huge function indicate an easy refactoring. Just extract the blocks within separated functions. Here, it would be `handle_contrast_files` for example.